internal/api/js/chrome/chrome.go

```diff
@@ -61,68 +52,30 @@ func MustRunAsyncFn[T any](t ct.TestLike, ctx context.Context, js string) *T {
 	return result
 }
 
-func RunHeadless(onConsoleLog func(s string), requiresPersistance bool, listenPort int) (*Browser, error) {
+func RunHeadless(onConsoleLog func(s string), listenPort int) (*Tab, error) {
+	// make a Chrome browser
 	browser, err := GlobalBrowser()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("GlobalBrowser: %s", err)
 	}
 
-	// Listen for console logs for debugging AND to communicate live updates
-	chromedp.ListenTarget(browser.Ctx, func(ev interface{}) {
-		switch ev := ev.(type) {
-		case *runtime.EventConsoleAPICalled:
-			for _, arg := range ev.Args {
-				s, err := strconv.Unquote(string(arg.Value))
-				if err != nil {
-					s = string(arg.Value)
-				}
-				onConsoleLog(s)
-			}
-		}
+	// Host the JS SDK
+	baseJSURL, closeSDKInstance, err := NewJSSDKWebsite(JSSDKInstanceOpts{
+		Port: listenPort,
 	})
-
-	// strip /dist so /index.html loads correctly as does /assets/xxx.js
-	c, err := fs.Sub(jsSDKDistDirectory, "dist")
 	if err != nil {
-		return nil, fmt.Errorf("failed to strip /dist off JS SDK files: %s", err)
+		return nil, fmt.Errorf("failed to create new js sdk instance: %s", err)
 	}
 
-	baseJSURL := ""
-	// run js-sdk (need to run this as a web server to avoid CORS errors you'd otherwise get with file: URLs)
-	var wg sync.WaitGroup
-	wg.Add(1)
-	mux := &http.ServeMux{}
-	mux.Handle("/", http.FileServer(http.FS(c)))
-	srv := &http.Server{
-		Addr:    fmt.Sprintf("127.0.0.1:%d", listenPort),
-		Handler: mux,
-	}
-	startServer := func() {
-		ln, err := net.Listen("tcp", srv.Addr)
-		if err != nil {
-			panic(err)
-		}
-		baseJSURL = "http://" + ln.Addr().String()
-		fmt.Println("JS SDK listening on", baseJSURL)
-		wg.Done()
-		srv.Serve(ln)
-		fmt.Println("JS SDK closing webserver")
-	}
-	go startServer()
-	wg.Wait()
```

Comment:
ugh it may be my covid-addled brain but I'm still struggling to follow this a bit. It still feels like this is mixing up refactoring and functional changes. Can't you make this easier for the reviewer to follow?

Sorry, I'm going to abort reviewing this, for now at least